src/spikeinterface/core/recording_tools.py

```diff
@@ -69,7 +69,7 @@ def _init_binary_worker(recording, file_path_dict, dtype, byte_offest, cast_unsi
 def write_binary_recording(
     recording: "BaseRecording",
     file_paths: list[Path | str] | Path | str,
-    dtype: np.ndtype = None,
+    dtype: np.dtype = None,
```

Comment:
For me it's okay as-is, unless we made a repo-wide decision to switch to proper numpy types e.g.` np.ndarray` -> `NDArray` (if my understanding that these two examples (dtype, ndarray) are equivalent is correct)

Reply:
Though out of interest, why `ndtype -> dtype`? Isn't `ndtype` preferred for typing?

Reply:
since this isn't technically the correct numpy typing I think either would be fine (they are both technically not right, no? ). I think a lot of our code uses np.array instead of np.ndarray for typing so I would say more for consistency than anything else.

>  np.ndarray -> NDArray

I'm not sure about this. the variable can actually be an np.ndarray. an argument can't be an np.dtype. It would be the object that is returned by `dtype='float'` would be `final_dtype = np.dtype('float')` so the argument isn't an np.dtype but the input into np.dtype. But to be honest I don't know the intricacies of numpy typing. I think this is why we all decided that using the constructor is fine even though it is not the actual object. Correct me if you understand this differently though.